agent_build/common.py

```diff
@@ -66,6 +67,46 @@ def recursively_delete_files_by_name(
                     break
 
 
+def recursively_delete_dirs_by_name(
+        root_dir: Union[str, pl.Path], *dir_names: str
+):
+    """
+    Deletes any directories that are in the current working directory or any of its children whose file names
+    match the specified regular expressions.
+
+    This will recursively examine all children of the current working directory.
+
+    If a directory is found that needs to be deleted, all of it and its children are deleted.
+
+    @param dir_names: A variable number of strings containing regular expressions that should match the file names of
+        the directories that should be deleted.
+    """
+
+    # Compile the strings into actual regular expression match objects.
+    matchers = []
+    for dir_name in dir_names:
+        matchers.append(re.compile(dir_name))
+
+    # Walk down the file tree, top down, allowing us to prune directories as we go.
+    for root, dirs, files in os.walk(root_dir):
+        # The list of directories at the current level to delete.
+        to_remove = []
+
+        # Examine all directories at this level, see if any get a match
+        for dir_path in dirs:
+            remove_it = False
+            for matcher in matchers:
+                if matcher.match(dir_path):
+                    remove_it = True
+            if remove_it:
+                to_remove.append(dir_path)
+
+        # Go back and delete it.  Also, remove it from dirs so that we don't try to walk down it.
+        for remove_dir_path in to_remove:
+            shutil.rmtree(os.path.join(root, remove_dir_path))
+            dirs.remove(remove_dir_path)
```

Comment:
I guess this works fine and doesn't throw "changed array during" iteration or similar error?

Reply:
This code just has been copied from the current `build_package.py` script, so it should work is it did before :)